chatterbot/adapters/input/variable_input_type_adapter.py

```diff
@@ -0,0 +1,60 @@
+from chatterbot.adapters.input import InputAdapter
+from chatterbot.utils.read_input import input_function
+from chatterbot.conversation import Statement
+
+
+JSON = 'json'
+TEXT = 'text'
+OBJECT = 'object'
+VALID_FORMATS = (JSON, TEXT, OBJECT, )
+
+
+class VariableInputTypeAdapter(InputAdapter):
+
+    def __init__(self, **kwargs):
+        super(VariableInputTypeAdapter, self).__init__(**kwargs)
+
+    def detect_type(self, statement):
+        input_type = type(statement)
+
+        if input_type is Statement:
```

Comment:
I'd recommend using `isinstance` over `type() is` here, to allow for subclasses to be detected properly.